ksy_style_guide.adoc

```diff
@@ -249,28 +248,36 @@ guesswork.
 
 * For simple non-repeated fields, use a simple singular form —
   i.e. `width`, `header`, `transaction_id`, `file`.
-* For repeated files (i.e. with `repeat: something`), use plular form
+* For a sequence (i.e. with `repeat: something`), use plular form
   — i.e. `files`, `transactions`.
 * Don't be overly verbose: use commonly understood abbreviations
   liberally, if it will improve readablity — i.e. `src_mac` or
   `src_mac_addr` instead of `source_media_access_control_address`
+* Repeated fileds which cannot be packed into a sequence should
+  have `id`s containing a number in the end. Numbers may have a
+  visually-obvious structure, like "the first digit is major, the second
+  one is minor".
 * For fields that are designed to be used to detect file type (AKA
-  "magic values"), use `magic` name, or, if there are several of them,
-  `magic1`, `magic2`, etc.
+  "magic values"), use `signature` or `magic` name, or, if there are
+  several of them, like `signature` or `magic0`, `magic1`, etc.
 * For reserved fields which are *known* to be unused, use `reserved`
-  name (or `reserved1`, `reserved2`, etc, if there are many of them)
+  name (or `reserved0`, `reserved1`, etc, if there are many of them)
 * For fields that designate *number / count* of something (in
-  particular, number of repetitions of some other structure), use
-  `num_` prefix and plular form — i.e. `num_questions`, `num_blocks`,
-  `num_nodes`
+  particular, number of repetitions of some other structure), use either
+  `_count` suffix or `count_of_` (what sounds better in your opinion)
+  prefix and a plular form — i.e. `count_of_questions`, `blocks_count`,
+  `nodes_count`
```

Comment:
Again, I don't think it's a good idea. The standard must be clear and avoid ambiguilities whenever it's possible. There is no point to make ksy developer to choose between several possible alternatives, especially using very subjective means (i.e. "sounds better"). The whole point of having a hard standard is to make it possible to use automated machinery to process stuff, and it's very hard to program a converter from `num_*` to `*_count` or `count_of_*`.

Also, both sound kinda quirky. Stuff like "blocks count", "nodes count" is plain wrong from English point of view. It should be "block count", "node count", etc, see [this SE entry](https://english.stackexchange.com/questions/349475/singular-or-plural-noun-preceeding-count) for examples. "Count of X" sounds like a title of some person for me (i.e. [count of Champagne](https://en.wikipedia.org/wiki/Count_of_Champagne), [count of Monte Cristo](https://en.wikipedia.org/wiki/The_Count_of_Monte_Cristo), [count of Barcelona](https://en.wikipedia.org/wiki/Count_of_Barcelona)).

Reply:
>There is no point to make ksy developer to choose between several possible alternatives, especially using very subjective means (i.e. "sounds better").

No point, but I guess we want to have readable names.

>The whole point of having a hard standard is to make it possible to use automated machinery to process stuff, and it's very hard to program a converter from num_* to *_count or count_of_*.

2 regexes with 1 capturing group each + one logical operation to recognize and parse such kind of names.

>Also, both sound kinda quirky. Stuff like "blocks count", "nodes count" is plain wrong from English point of view. It should be "block count", "node count", etc, see this SE entry for examples.

You are right. But you can also think about this as about ```count``` property of ```blocks``` with a ```_``` instead of ```.```, since ```.``` is not allowed.

>"Count of X" sounds like a title of some person for me (i.e. count of Champagne, count of Monte Cristo, count of Barcelona).

IMHO it's completely OK since there isn't many counts, dukes and baronets in programming.

Reply:
You're kind of contradicting yourself:

> No point, but I guess we want to have readable names.
> But you can also think about this as ...

Either you want "normal English" names, then you can't have "blocks count" and "count of blocks". If you allow some special naming, then "num_blocks" is about as readable as your versions. Hungarian apps notation recommends "nBlocks" for these purposes, and even that is understandable by majority of people.

Reply:
I especially dislike ```num``` prefix because of its redundancy and ambiguity. It says that the field is number, but we see that it is a number from the type. It says nothing about the purpose of that number. ```cnt``` should be better here. And again, I'm against prefixes.

>Hungarian notation

is evil.